feat: Этап 2. Реализовать основной сервис

elayhan opened the pull request.

elayhan committed:
feat: Этап 2. Реализовать основной сервис

elayhan committed:
fix: conflict при попытке записи уже существующей категории

afomkina commented on ewm-main-service/src/main/java/ru/practicum/category/model/Category.java:
```diff
@@ -0,0 +1,18 @@
+public class Category {
+    @Id
+    @GeneratedValue(strategy = GenerationType.IDENTITY)
+    Long id;
```
Не хватает модификаторов доступа у полей

afomkina commented on ewm-main-service/src/main/java/ru/practicum/compilation/model/Compilation.java:
```diff
@@ -0,0 +1,31 @@
+public class Compilation {
+    @Id
+    @GeneratedValue(strategy = GenerationType.IDENTITY)
+    Long id;
```
Не хватает модификаторов доступа у полей

afomkina commented on ewm-main-service/src/main/java/ru/practicum/event/model/location/Location.java:
```diff
@@ -0,0 +1,18 @@
+public class Location {
+    @Id
+    @GeneratedValue(strategy = GenerationType.IDENTITY)
+    Long id;
```
Не хватает модификаторов доступа

afomkina commented on ewm-main-service/src/main/java/ru/practicum/event/repository/location/LocationRepository.java:
```diff
@@ -0,0 +1,7 @@
+import org.springframework.data.jpa.repository.JpaRepository;
+import ru.practicum.event.model.location.Location;
+
+public interface LocationRepository extends JpaRepository<Location, Long> {
```
Не похоже, что мы его где-то используем

afomkina commented on ewm-main-service/src/main/java/ru/practicum/event/service/EventServiceImpl.java:
```diff
@@ -0,0 +1,326 @@
+        strategyMap.get(state).accept(state);
+    }
+
+    private <T extends UpdateEventDto> void updateEvent(Event event, T updateEvent) {
```
Обновление полей тоже лучше делегировать мапстракту, так получится сильно сократить код. Чтобы поля с null значениениями не маппились, можно использовать [данный подход](https://stackoverflow.com/questions/54174437/mapstruct-how-to-set-different-null-strategy-for-different-mapping-methods)

afomkina commented on ewm-main-service/src/main/java/ru/practicum/event/service/EventServiceImpl.java:
```diff
@@ -0,0 +1,326 @@
+            rangeEnd = LocalDateTime.now().plusYears(30);
+        }
+
+        if (rangeEnd.isBefore(rangeStart)) {
```
rangeStart может быть равен null, можем упасть с NPE

afomkina commented on ewm-main-service/src/main/resources/schema.sql:
```diff
@@ -0,0 +1,68 @@
+  CONSTRAINT event_participant_limit_positive_chk CHECK (participant_limit >= 0),
+  constraint event_date_after_create_chk check (event_date + interval '2' hour >= created_on)
+);
+create index if not exists event_lower_annotation_idx on event (lower(annotation));
```
🔥

afomkina commented on ewm-main-service/src/main/java/ru/practicum/util/StatUtil.java:
```diff
@@ -0,0 +1,55 @@
+
+@Service
+@RequiredArgsConstructor
+public class StatUtil {
```
Это не совсем утильный класс, а самый настоящий сервис, который обращается к клиенту статистики, нужно переименовать

afomkina reviewed (changes requested):
У тебя получилась очень клевая работа, молодец 👍

elayhan committed:
fix: Исправить замечания ревью

fix: Добавить модификаторы доступа entity-классам
fix: Исключить NPE при запросе событий(граница переода может быть null)
feat: Делегировать обновление событий мапструкт

afomkina commented on ewm-main-service/src/main/java/ru/practicum/event/mapper/EventMapper.java:
```diff
@@ -40,4 +41,8 @@ public interface EventMapper {
 
     List<EventFullDto> eventListToEventFullDto(List<Event> eventList);
 
+    @BeanMapping(nullValuePropertyMappingStrategy = NullValuePropertyMappingStrategy.IGNORE,
```
🔥
